test_pytest_rerunfailures.py

```diff
@@ -267,3 +267,42 @@ def test_fail_two():
     time.sleep.assert_called_with(delay_time)
 
     assert_outcomes(result, passed=0, failed=1, rerun=2)
+
+
+def test_rerun_on_class_setup_with_first_execution_error_for_parametrized_test(testdir):
+    testdir.makepyfile("""
+        import pytest
+        
+        class TestFoo(object):
+            execution_number = 1
+            @classmethod
+            def setup_class(cls):
+                if cls.execution_number:
+                    cls.execution_number -= 1
+                    assert False
+                assert True
+            @pytest.mark.parametrize('param', [1, 2, 3])
+            def test_pass(self, param):
+                assert param""")
+    result = testdir.runpytest('--reruns', '1')
+    assert_outcomes(result, passed=3, rerun=1)
+
+
+def test_rerun_on_class_scope_fixture_setup_with_first_execution_error_for_parametrized_test(testdir):
```

Comment:
The function names of these tests are quite long. Could you please shorten them and write the more verbose description in a docstring? This reminds me to add a flake8 test to the tox run.

Reply:
I will be really appreciated if you could suggest me the names for these cases. I really cannot find any clearer and shorter test names for this specific contexts :)